File: demo/app/app/src/main/java/com/magicvector/demo/manager/CatalogManager.kt
package com.magicvector.demo.manager

import android.content.Context
import android.content.Intent
import com.magicvector.demo.activity.ListActivity
import com.magicvector.demo.activity.NetworkActivity
import com.magicvector.demo.domain.vo.CatalogItem

object CatalogManager {

    fun getCatalogItems(): List<CatalogItem> {
        return listOf(
            CatalogItem(
                id ="1",
                title = "网络",
                subtitle = "网络相关Demo",
                cls = NetworkActivity::class.java
            ),
            CatalogItem(
                id ="2",
                title = "My列表",
                subtitle = "自定义compose列表相关Demo",
                cls = ListActivity::class.java
            ),
        )
    }

    fun onItemClick(item: CatalogItem, activity: Context): OnClickCatalogItem{
        val intent = Intent(
            activity,
            item.cls
        )
        return object : OnClickCatalogItem {
            override fun onClick(item: CatalogItem) {
                activity.startActivity(intent)
            }
        }
    }

}

File: demo/flutternew/lib/manager/CatalogManager.dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:flutternew/config/app_route.dart';

import '../domain/vo/CatalogItem.dart';
import '../l10n/app_localizations.dart';



typedef OnClickCatalogItem = void Function(CatalogItem item);

class CatalogManager {
  // 模拟获取目录数据
  static List<CatalogItem> getCatalogItems(AppLocalizations l10n) {
    return [
      CatalogItem(
        id: "1",
        title: l10n.network,
        subtitle: "network demo",
        iconRes: Icons.network_wifi,
        routeName: AppRoutes.network,
      ),
      CatalogItem(
        id: "2",
        title: l10n.chat,
        subtitle: "chatList demo",
        iconRes: CupertinoIcons.chat_bubble,
        routeName: AppRoutes.chat,
      ),
      CatalogItem(
        id: "3",
        title: l10n.native,
        subtitle: "native demo",
        iconRes: Icons.android_sharp,
        routeName: AppRoutes.native,
      ),
      CatalogItem(
        id: "4",
        title: l10n.xfyun_stt,
        subtitle: "科大讯飞语音识别",
        iconRes: Icons.mic,
        routeName: AppRoutes.xfyunStt,
      ),
    ];
  }

  // 处理Item点击跳转逻辑
  static void onItemClick(CatalogItem item, BuildContext context) {
    if (item.routeName != null && item.routeName!.isNotEmpty) {
      print('将要跳转到${item.routeName}');
      Navigator.pushNamed(context, item.routeName!);
    } else {
      // 如果没有设置路由，显示提示
      _showFeatureComingSoon(context, item.title);
    }
  }


  // 显示功能开发中提示
  static void _showFeatureComingSoon(BuildContext context, String title) {
    ScaffoldMessenger.of(context).showSnackBar(
      SnackBar(
        content: Text('$title 功能正在开发中'),
        duration: const Duration(seconds: 2),
        action: SnackBarAction(
          label: '知道了',
          onPressed: () {},
        ),
      ),
    );
  }

}